onyo/lib/filters.py

```diff
@@ -61,6 +61,8 @@ def match(self, asset: dict) -> bool:
                 asset[self.key] is None or asset[self.key] == ''):
             return True
         elif self.key not in asset.keys() or self.value == UNSET_VALUE:
+            # Note: This is weird.
```

Comment:
Right, this shouldn't be in here, will fix that in a separate PR.
However, to clarify this:

- your second read is correct: The code amounts to an XOR and that's OK, it just reads weird and therefore isn't nice to maintain
- what it handles are two cases:
  - The key we are looking for is not in the asset and we are looking for a value other than `<unset>`. 
  - The key actually is present, but we are looking for `<unset>`.

Reply:
Ok. The comment referencing the previous `elif` statements made me think that the check was intentionally a complex intersection of the previous statements and current one.

Since the statements each `return`, there's no performance gained by using `elif`. In the PR you said you'll open, I propose changing these to `if` statements, so that one does not need to mentally load all four in order to understand the last one.